easybuild/easyblocks/m/mxnet.py

```diff
@@ -0,0 +1,222 @@
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for MXNet, implemented as an easyblock
+
+@author: Ward Poelmans (Free University of Brussels)
+"""
+import glob
+import os
+import shutil
+from distutils.version import LooseVersion
+
+from easybuild.easyblocks.generic.makecp import MakeCp
+from easybuild.easyblocks.generic.pythonpackage import PythonPackage
+from easybuild.easyblocks.generic.rpackage import RPackage
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import apply_regex_substitutions, copy_file, change_dir, mkdir
+from easybuild.tools.filetools import rmtree2, symlink, write_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_MXNet(MakeCp):
+    """Easyblock to build and install MXNet"""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Change default values of options"""
+        extra = MakeCp.extra_options()
+        # files_to_copy is not mandatory here
+        extra['files_to_copy'][2] = CUSTOM
+        return extra
+
+    def __init__(self, *args, **kwargs):
+        """Initialize custom class variables."""
+        super(EB_MXNet, self).__init__(*args, **kwargs)
+
+        self.mxnet_src_dir = None
+        self.py_ext = PythonPackage(self, {'name': self.name, 'version': self.version})
+        self.py_ext.module_generator = self.module_generator
+        self.r_ext = RPackage(self, {'name': self.name, 'version': self.version})
+        self.r_ext.module_generator = self.module_generator
+
+    def extract_step(self):
+        """
+        Prepare a combined MXNet source tree. Move all submodules
+        to their right place.
+        """
+        # Extract everything into separate directories.
+        super(EB_MXNet, self).extract_step()
+
+        mxnet_dirs = glob.glob(os.path.join(self.builddir, 'mxnet-*'))
+        if len(mxnet_dirs) == 1:
+            self.mxnet_src_dir = mxnet_dirs[0]
+            self.log.debug("MXNet dir is: %s", self.mxnet_src_dir)
+        else:
+            raise EasyBuildError("Failed to find/isolate MXNet source directory: %s", mxnet_dirs)
+
+        for srcdir in os.listdir(self.builddir):
+            if not srcdir.startswith('mxnet-'):
```

Comment:
collapse to:

```python
for srcdir in [d for d in os.listdir(self.builddir) if not d.startswith('mxnet-')]:
```

since you're using `os.listdir` here, better wrap the whole thing in `try`/`except (IOError, OSError) as err`?